### PR TITLE
Feature/idenstatekzflow

### PR DESCRIPTION
**Pull request by ed255**
The main part of this PR is adding a function to calculate the zk proof for identity state updates, but along the way I also updated some minor (but related) stuff:

- Add function to calculate zk proof for iden state update
  - Test Issuer.GenZkProofIdenStateUpdate
- Add function to download idenstate zk files
- Update export in keystore, don't use capitals in errors
- Refactor elems to big ints conversions

### CODE REVIEW

File: keystore/keystore.go [hunk 1]
```diff
@@ -47,8 +47,11 @@ const (
 )
 
 var (
-	ErrStorageLock     = fmt.Errorf("Unable to acquire storage lock")
-	ErrStorageUnlocked = fmt.Errorf("Storage is not locked")
+	ErrStorageLock     = fmt.Errorf("unable to acquire storage lock")
+	ErrStorageUnlocked = fmt.Errorf("storage is not locked")
+	ErrKeyNotInCache   = fmt.Errorf("public key not found in the cache.  Maybe it's not unlocked")
+	ErrKeyNotFound     = fmt.Errorf("public key not found in the key store")
+	ErrInvalidEncData  = fmt.Errorf("Invalid encrypted data")
```
Comment (arnaucube) [inline keystore/keystore.go:54]: If the commit says `don't use capitals in errors` maybe the ErrInvalidEncData should be in lowercase also?

File: identity/issuer/issuer.go [hunk 2]
```diff
@@ -22,12 +25,20 @@ import (
...
+	zktypes "github.com/iden3/go-circom-prover-verifier/types"
+	witnesscalc "github.com/iden3/go-circom-witnesscalc"
+	cryptoUtils "github.com/iden3/go-iden3-crypto/utils"
+	wasm3 "github.com/iden3/go-wasm3"
+
 	log "github.com/sirupsen/logrus"
 )
 
 var (
 	ErrIdenGenesisOnly           = fmt.Errorf("Identity is genesis only")
```
Comment (arnaucube) [inline identity/issuer/issuer.go:39]: From the previous commit, I understand that the idea is to use lowercase for errors

File: keystore/keystore.go (hunk 1 above)
Comment (ed255) [reply keystore/keystore.go]: thanks for spotting this!

File: identity/issuer/issuer.go (hunk 2 above)
Comment (ed255) [reply identity/issuer/issuer.go]: Yes! updated.

**Pull request merged**